### PR TITLE
modem: wncm14a2a: Update driver to use gpio_dt_spec

### PR DESCRIPTION
**Pull request by galak**
Simplify the implementation by using gpio_dt_spec.

Signed-off-by: Kumar Gala <galak@kernel.org>

### CODE REVIEW

File: drivers/modem/wncm14a2a.c [hunk 1]
```diff
@@ -1283,9 +1240,8 @@ static int modem_pin_init(void)
 	 * When enabled, there will be no changes in the above 4 pins...
 	 */
 	LOG_DBG("SIG_TRANS_ENA_PIN -> ENABLED");
-	gpio_pin_set_raw(ictx.gpio_port_dev[SHLD_3V3_1V8_SIG_TRANS_ENA],
-			 pinconfig[SHLD_3V3_1V8_SIG_TRANS_ENA].pin,
-			 SHLD_3V3_1V8_SIG_TRANS_ENABLED);
+	gpio_pin_set_dt(&wncm14a2a_cfg.gpio[SHLD_3V3_1V8_SIG_TRANS_ENA],
+			SHLD_3V3_1V8_SIG_TRANS_ENABLED);
```
Comment (gmarull) [inline drivers/modem/wncm14a2a.c:1244]:
```suggestion
	gpio_pin_set_dt(&wncm14a2a_cfg.gpio[SHLD_3V3_1V8_SIG_TRANS_ENA], 1);
```

suggestion: use plain `1`/`0`

**Review by rerickson1 — APPROVED** [review]

File: drivers/modem/wncm14a2a.c (hunk 1 above)
Comment (rerickson1) [reply drivers/modem/wncm14a2a.c:1244]: I like having the defines. It makes it easier for someone who has to change the IO logic. Maybe their design level 0 is enabled and now they only have to change the define.

Comment (gmarull) [reply drivers/modem/wncm14a2a.c:1244]: > I like having the defines. It makes it easier for someone who has to change the IO logic. Maybe their design level 0 is enabled and now they only have to change the define.

If level has to be controlled by the driver, `_raw` should be used. These defines can have different results depending on gpio flags in DT.

Comment (rerickson1) [reply drivers/modem/wncm14a2a.c:1244]: so for my learning, `gpio_pin_set_dt(&wncm14a2a_cfg.gpio[SHLD_3V3_1V8_SIG_TRANS_ENA], 1);` doesn't just mean set the pin high, it depends on what is configured in the DT? So setting the pin could be setting it to 0 if that's how the DT is setup? If that's the case, then I am all for leaning on DT config to do this properly. That would mean any DT in the tree needs to be looked at to make sure it is setup correctly so this change does break operation.

**Review by rerickson1 — CHANGES_REQUESTED** [review]
Change active high/low and GPIO flags in boards/shields/wnc_m14a2a/boards/frdm_k64f.overlay

**Comment by galak** [comment]
> Change active high/low and GPIO flags in boards/shields/wnc_m14a2a/boards/frdm_k64f.overlay

how do you recommend I set them?

**Review by rerickson1 — APPROVED** [review]
Looks good

**Review by mniestroj — APPROVED** [review]

**Pull request merged**